cms/wizards/wizard_pool.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+
+from django.utils.translation import ugettext as _
+
+from cms.utils.django_load import load
+
+from .wizard_base import Wizard
+
+
+class AlreadyRegisteredException(Exception):
+    pass
+
+
+class WizardPool(object):
+    _entries = {}
+    _discovered = False
+
+    def __init__(self):
+        self._reset()
+
+    # PRIVATE METHODS -----------------
+
+    def _discover(self):
+        if not self._discovered:
+            load('cms_wizards')
+            self._discovered = True
+
+    def _clear(self):
+        """Simply empties the pool but does not clear the discovered flag."""
+        self._entries = {}
+
+    def _reset(self):
+        """Clears the wizard pool and clears the discovered flag."""
+        self._clear()
+        self._discovered = False
+
+    # PUBLIC METHODS ------------------
+
+    @property
+    def discovered(self):
+        """
+        A public getter for the private property _discovered. Note, there is no
+        public setter.
+        """
+        return self._discovered
+
+    def is_registered(self, entry):
+        """
+        Returns True if the provided entry is registered.
+
+        NOTE: This method triggers pool discovery.
+        """
+        self._discover()
+        return entry.id in self._entries
+
+    def register(self, entry):
+        """
+        Registers the provided «entry».
+
+        Raises AlreadyRegisteredException if the entry is already registered.
+
+        NOTE: This method triggers pool discovery.
+        """
+        assert isinstance(entry, Wizard), u"entry must be an instance of Wizard"
+        if self.is_registered(entry):
```

Comment:
I don't think we should call is_registered() here because it could result in some sort of infinite loop.
when registering an item we shouldn't trigger the discovery